### ISSUE TITLE
422 errors not propagated to UX

### ISSUE THREAD

**Issue by llunaCreixent** [CirclesUBI/circles-myxogastria#547]
**Describe the bug**
UX shows that I have transferred some Circles to an account, but actually in the Network tab I see a 422 error and the transaction will never succeed.

**To Reproduce**
1. Go to circles.garden
2. Select an account that is already migrated to 1.3.0
3. Open the Network tab in your browser
4. Send some circles to anyone
5. See popup of successful transfer in UX
6. See the error in the Network tab (422), and if you check in blockscout, the transaction never succeeds

**Expected behavior**
In the UX I expect not seeing that my transfer is successful

**Comment by llunaCreixent** [comment]
Please take into account in which user stories the errors are produced, because maybe not always is necessary to show the user that there was an error.

The user story where I found the issue is: an account that has not migrated to Safe v1.3.0 tries to do a transaction in circles.garden (with the myxogastria version that doesn't support the Safe v1.3.0 yet) @mikozet 

I'm mentioning this because it's a good idea to verify is this issue happens in other cases.

**Comment by mikozet** [comment]
In SendConfirm.js 

This is a place where the problem occurs.

```
const handleSend = async () => {
    dispatch(showSpinnerOverlay());
    setIsConfirmationShown(false);

    try {
      await dispatch(transfer(address, amount, paymentNote));

      dispatch(
        notify({
          text: translate('SendConfirm.successMessage', {
            amount,
            username: receiver,
          }),
          type: NotificationsTypes.SUCCESS,
        }),
      );

      setIsSent(true);
    } catch (error) {
      logError(error);
      let text;

      if (error instanceof TransferError) {
        // Convert TransferError codes into human readable error messages
        let messageId = 'Unknown';
        if (error.code === ErrorCodes.TOO_COMPLEX_TRANSFER) {
          messageId = 'TooComplex';
        } else if (error.code === ErrorCodes.INVALID_TRANSFER) {
          messageId = 'Invalid';
        } else if (error.code === ErrorCodes.TRANSFER_NOT_FOUND) {
          messageId = 'NotFound';
        }
        text = translate('SendConfirm.errorMessageTransfer' + messageId, {
          amount,
          username: receiver,
        });
      } else {
        text = translate('SendConfirm.errorMessage', {
          errorMessage: formatErrorMessage(error),
        });
      }

      dispatch(
        notify({
          text,
          type: NotificationsTypes.ERROR,
        }),
      );
    }

    dispatch(hideSpinnerOverlay());
  };
 ```

For some reason we dispatch a message even if error occurs..
```
dispatch(
        notify({
          text: translate('SendConfirm.successMessage', {
            amount,
            username: receiver,
          }),
          type: NotificationsTypes.SUCCESS,
        }),
      );
 ```

**Comment by mikozet** [comment]
It looks like 422 which is coming from the server is treated as a valid response and that's why we don't enter catch block and display success message. Maybe it might be worth to check in `try` block if 422 is coming and then react accordingly..

it might be related also with this comment where similar problem occurs:
https://github.com/CirclesUBI/circles-myxogastria/pull/581/files#r1087525629

**Comment by louilinn** [comment]
related to behaviour in https://github.com/CirclesUBI/circles-myxogastria/issues/589

**Comment by mikozet** [comment]
As talked with @JacqueGM we should wait for replacement of pathfinder service first

**Comment by louilinn** [comment]
I think this has been resolved now? I have seen a 422 in the console when a transaction fails and gotten the error notification and no success.

what do you think @llunaCreixent? was it resolved when we do the retry logic in transfer and throwing an error if it doesnt reach success?

![Screenshot from 2023-04-13 12-31-20.png](https://images.zenhubusercontent.com/63494aa5f1209367413ba666/c97384b2-a792-4b67-8b19-83a53ad65ab6)n
![Screenshot from 2023-04-13 12-26-34.png](https://images.zenhubusercontent.com/63494aa5f1209367413ba666/e25c860f-cb68-444c-9d3b-f8115165749a)

**Comment by llunaCreixent** [comment]
> I think this has been resolved now? I have seen a 422 in the console when a transaction fails and gotten the error notification and no success.
> 
> what do you think @llunaCreixent? was it resolved when we do the retry logic in transfer and throwing an error if it doesnt reach success?
> 
> ![Screenshot from 2023-04-13 12-31-20.png](https://images.zenhubusercontent.com/63494aa5f1209367413ba666/c97384b2-a792-4b67-8b19-83a53ad65ab6)n
> ![Screenshot from 2023-04-13 12-26-34.png](https://images.zenhubusercontent.com/63494aa5f1209367413ba666/e25c860f-cb68-444c-9d3b-f8115165749a) 

Yes, it was resolved with your PR

**Comment by louilinn** [comment]
Solved by #581
